src/hierarchical_deterministic/bip39/bip39_seed.rs

```diff
@@ -0,0 +1,106 @@
+use crate::{decl_secret_bytes, prelude::*};
+
+decl_secret_bytes!(
+    /// A BIP39 seed for hierarchal deterministic wallets, as per the [BIP39 standard][doc].
+    ///
+    /// We typically obtain this by calling [`to_seed` on `MnemonicWithPassphrase`][MnemonicWithPassphrase::to_seed].
+    ///
+    /// [doc]: https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#user-content-From_mnemonic_to_seed
+    BIP39Seed,
+    64
+);
+
+impl HDPath {
+    fn hardened_chain(&self) -> Vec<IotaSlip10PathComponent> {
+        self.components
+            .iter()
+            .map(|c| c.value)
+            .map(|v| IotaSlip10PathComponent::try_from(v).expect("Should work"))
+            .collect_vec()
+    }
+}
+
+use crypto::{
+    keys::slip10::{self as IotaSlip10, Hardened as IotaSlip10PathComponent},
+    signatures::ed25519 as IotaSlip10Ed25519,
+    signatures::secp256k1_ecdsa as IotaSlip10Secp256k1,
+};
+
+impl BIP39Seed {
+    fn derive_slip10_private_key<K, I>(&self, chain: I) -> IotaSlip10::Slip10<K>
+    where
+        K: IotaSlip10::IsSecretKey
+            + IotaSlip10::WithSegment<<I as Iterator>::Item>,
+        I: Iterator,
+        <I as Iterator>::Item: IotaSlip10::Segment,
+    {
+        let iota_seed = IotaSlip10::Seed::from_bytes(&*self.0);
+        iota_seed.derive(chain)
+        // `IotaSlip10::Seed` implements `ZeroizeOnDrop` so should now be zeroized.
```

Comment:
Thx to feature flag `zeroize` in IOTA crypto crates all IOTA types are now Zeroize and ZeroizeOnDrop.